test/integration/checkout_integration_test.go

```diff
@@ -159,3 +159,92 @@ func TestForcingACleanCheckout(t *testing.T) {
 	tester.MustMock(t, "rm").Expect("-rf", tester.CheckoutDir())
 	tester.RunAndCheck(t, "BUILDKITE_CLEAN_CHECKOUT=true")
 }
+
+func TestCheckoutOnAnExistingRepositoryWithoutAGitFolder(t *testing.T) {
+	tester, err := NewBootstrapTester()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer tester.Close()
+
+	// Create an existing checkout
+	out, err := tester.Repo.Execute("clone", "-v", "--", tester.Repo.Path, tester.CheckoutDir())
+	if err != nil {
+		t.Fatalf("Clone failed with %s", out)
+	}
+
+	if err = os.RemoveAll(filepath.Join(tester.CheckoutDir(), ".git")); err != nil {
+		t.Fatal(err)
+	}
+
+	agent := tester.MustMock(t, "buildkite-agent")
+	agent.
+		Expect("meta-data", "exists", "buildkite:git:commit").
+		AndExitWith(0)
+
+	tester.RunAndCheck(t)
+}
+
+func TestCheckoutOnABrokenExistingRepository(t *testing.T) {
+	t.Skipf("This is currently broken")
```

Comment:
This meant to be skipped?

Reply:
Only in that we don't currently recover from it. I write the tests expecting that we would, and then figured out that `git rev-parse` won't detect corrupted objects.

Reply:
I figure having a skipped test reminds us that we don't support it, but it could possibly just be removed.

Reply:
Oh right. I thought `"This is currently broken"` meant that the test was currently broken! Perhaps that just needs to be clarified with your words above.